scan/scan_common.go

```diff
@@ -163,57 +170,72 @@ func (fs FamilySet) RunScans(host, ip, family, scanner string, dur time.Duration
 		return nil, err
 	}
 
-	results := make(chan map[string]FamilyResult)
-	timeout := make(chan bool)
-
-	if dur > 0 {
-		go startTimer(dur, timeout)
-	}
+	results := make(chan *Result)
 
-	familyResults := make(map[string]FamilyResult)
+	var familyWG sync.WaitGroup
+	familyWG.Add(len(fs))
+	go func() {
+		familyWG.Wait()
+		close(results)
```

Comment:
Maybe a comment would be good here. This goroutine is spawned to wait for all the scans to finish and then signal this by closing the (already returned) `results` channel